File: 151.js
// 151. Reverse Words in a String


/**
 * @param {string} s
 * @return {string}
 */
var reverseWords = function(s) {

    let trimmedS = s.trim()
    let reversedS = []
    let words = []
    let temp = ""

    for (let i=0; i<trimmedS.length; i++) {
        
        if (/[a-zA-Z0-9]/.test(trimmedS[i])) {
            temp += trimmedS[i]
            if (i === trimmedS.length-1) {
                words.push(temp)
            }
        } else if (trimmedS[i] === " ") {
            if (!temp) {
                continue
            } else {
                words.push(temp)
                temp = ""
            }     
        }
    }
    
    reversedS = words.reverse()
    return reversedS.join(" ")   
};

File: 151.py
# 151. Reverse Words in a String

class Solution(object):
    def reverseWords(self, s):
        """
        :type s: str
        :rtype: str
        """
        
        trimmed_s = s.strip()

        splitted_s = trimmed_s.split()

        reversed_s = splitted_s[::-1] # use slicing to reverse the string
        # reversed_s = reversed(splitted_s) # use the iterator (reversed) to reverse the string

        joined_s = " ".join(reversed_s)

        return joined_s